lib/client/signalflow/signalflow_client.js

```diff
@@ -63,28 +62,78 @@ function SignalflowClient(apiToken, options) {
         return metaDataMap[tsid];
       },
       stream: function (fn, errorFn) {
-        if (typeof fn !== 'function') {
+        if (typeof fn !== "function") {
           return false;
         }
         callback = fn;
         msgBuffer.forEach(resolveMessage);
         msgBuffer = [];
         return true;
-      }
+      },
     };
   }
 
+  function liveTailRequest(opts) {
+    var msgBuffer = [];
+    var callback = null;
+
+    function resolveMessage(msg) {
+      // we're duck typing error messages here, but it should be okay.
+      if (msg.hasOwnProperty("error") && !msg.type) {
+        errorMessageCallback(msg);
+      } else {
+        callback(null, msg);
+      }
+    }
+
+    function errorMessageCallback(msg) {
+      callback(msg, null);
+    }
+
+    function msgCallback(msg) {
+      if (!callback) {
+        msgBuffer.push(msg);
+      } else {
+        resolveMessage(msg);
+      }
+    }
+
+    var callbacks = {
+      onMessage: msgCallback,
+      onError: errorMessageCallback,
+    };
+
+    var requestId = rm.registerLiveTail(opts, callbacks, null);
+
+    return {
+      close: function () {
+        return rm.stopLiveTail(requestId);
+      },
+      stream: function (fn, errorFn) {
+        if (typeof fn !== "function") {
+          return false;
+        }
+        callback = fn;
+        msgBuffer.forEach(resolveMessage);
+        msgBuffer = [];
+        return true;
+      },
+    };
+  }
   return {
     disconnect: disconnect,
     execute: function (opts) {
-      return signalflowRequest(opts, 'execute');
+      return signalflowRequest(opts, "execute");
     },
     explain: function (opts) {
-      return signalflowRequest(opts, 'explain');
+      return signalflowRequest(opts, "explain");
     },
     preflight: function (opts) {
-      return signalflowRequest(opts, 'preflight');
-    }
+      return signalflowRequest(opts, "preflight");
+    },
+    liveTail: function (opts) {
```

Comment:
You can change that to simply:

```js
  liveTail: liveTailRequest,
```

Reply:
ok I did that, but looks like I need to send in some opts? Because addComputation needs to require these 

`  var requiredParams = {
    execute: ['program'],
    explain: ['incidentId']
  };`

Reply:
where do these come from?

Reply:
Those are for signalflow computation; you don't want to go through this path. Don't use `addComputation()` directly.